File: labs/lab5_6/Lab2_GeometricFigures/Rectangle.cs
using System;

namespace Lab2_GeometricFigures
{
    public class Rectangle : GeometricFigure, IPrint
    {
        public double Width { get; set; }
        public double Height { get; set; }

        public Rectangle(double width, double height)
        {
            if (width <= 0 || height <= 0)
                throw new ArgumentException("Ширина и высота должны быть положительными числами");

            Width = width;
            Height = height;
        }

        public override double Area()
        {
            return Width * Height;
        }

        public override string ToString()
        {
            return $"Прямоугольник: Ширина = {Width:F2}, Высота = {Height:F2}, Площадь = {Area():F2}";
        }

        public void Print()
        {
            Console.WriteLine(ToString());
        }
    }
}

File: labs/lab2/lab_python_oop/Rectangle.py
from lab_python_oop.GeometricFigure import GeometricFigure
from lab_python_oop.FigureColor import Color

class Rectangle(GeometricFigure):
    name = "Прямоугольник"

    def __init__(self, width, height, color):
        self.width = width
        self.height = height
        self.colorObj = Color(color)

    def area(self):
        return self.width * self.height

    def __repr__(self):
        return "Фигура: {0}, ширина: {1}, высота: {2}, цвет: {3}, площадь: {4}".format(
            self.getName(), self.width, self.height, self.colorObj, self.area())
